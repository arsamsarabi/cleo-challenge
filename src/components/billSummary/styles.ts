import styled, { css } from 'styled-components'

import { ImageWithFallback } from '../imageWithFallback/ImageWithFallback'

export const Wrapper = styled.div<WithTheme>(({ theme: { measurements } }) => {
  return css`
    display: flex;
    justify-content: flex-start;
    align-items: center;
    flex: 1;
    padding: ${measurements.medium};

    &:hover {
      cursor: pointer;
    }

    svg {
      margin: 0 ${measurements.large} 0 auto;
    }
  `
})

export const MerchantImage = styled(ImageWithFallback)<WithTheme>(
  ({ theme: { measurements } }) => {
    return css`
      width: 75px;
      height: 75px;
      margin-right: ${measurements.large};
    `
  }
)

export const CategoryIcon = styled.img(() => {
  return css`
    width: 20px;
    height: 20px;
  `
})

export const MerchantNameRow = styled.div<WithTheme>(
  ({ theme: { measurements, palette } }) => {
    return css`
      display: flex;
      h1 {
        font-size: 1.25rem;
        color: ${palette.grey[800]};
        margin-bottom: ${measurements.medium};
        margin-left: ${measurements.small};
      }
    `
  }
)

export const TransactionSummary = styled.p<WithTheme>(
  ({ theme: { measurements, palette } }) => {
    return css`
      font-size: 0.9rem;
      color: ${palette.grey[700]};
      span {
        font-weight: bold;
        &:first-of-type {
          margin-right: ${measurements.xsmall};
        }
        &:last-of-type {
          margin-left: ${measurements.xsmall};
        }
      }
    `
  }
)
