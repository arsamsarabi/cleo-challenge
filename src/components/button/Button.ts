import styled, { css } from 'styled-components'

export type ButtonColorMode = 'normal' | 'danger' | 'success'

export type ActionButtonProps = WithTheme & {
  mode?: ButtonColorMode
}

export const ActionButton = styled.button<ActionButtonProps>(
  ({ theme: { measurements, palette }, mode = 'normal' }) => {
    const colorMap: { [key in typeof mode]: string } = {
      normal: palette.blue,
      success: palette.common.green,
      danger: palette.common.red,
    }

    return css`
      outline: none;
      border: none;
      color: ${colorMap[mode]};
      background-color: ${palette.common.white};
      padding: ${measurements.small} ${measurements.medium};
      min-height: 40px;
      min-width: 140px;
      text-transform: uppercase;
      border: 2px solid ${colorMap[mode]};
      border-radius: 4px;
      transition: 0.2s all ease-in-out;
      display: flex;
      justify-content: flex-start;
      align-items: center;

      svg {
        color: ${colorMap[mode]};
        margin-right: ${measurements.medium};
      }

      &:hover {
        cursor: pointer;
        color: ${palette.common.white};
        background-color: ${colorMap[mode]};
        font-weight: bold;
        border-color: ${colorMap[mode]};

        svg {
          color: ${palette.common.white};
        }
      }
    `
  }
)
