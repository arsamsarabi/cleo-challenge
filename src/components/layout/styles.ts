import styled, { css } from 'styled-components'

export const Wrapper = styled.main(() => {
  return css`
    width: 100vw;
    height: 100vh;
    overflow: hidden;
    display: flex;
  `
})

export const MainContent = styled.section(
  ({ theme: { measurements, palette } }: WithTheme) => {
    return css`
      box-shadow: -10px 0px 10px 1px ${palette.grey[500]};
      border-top-left-radius: 16px;
      border-bottom-left-radius: 16px;
      padding: ${measurements.large};
      background-color: ${palette.grey[50]};
      flex: 1;
      display: flex;
      flex-direction: column;
      overflow-y: auto;
    `
  }
)
