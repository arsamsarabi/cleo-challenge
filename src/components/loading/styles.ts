import styled, { css } from 'styled-components'

export const Wrapper = styled.div(() => {
  return css`
    flex: 1;
    width: 100%;
    min-height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
  `
})

export const LoadingGif = styled.img(() => {
  return css`
    width: 160px;
  `
})
