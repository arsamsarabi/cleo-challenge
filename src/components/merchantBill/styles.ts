import styled, { css } from 'styled-components'
import { rgba } from 'polished'

export const Wrapper = styled.div<WithTheme>(
  ({ theme: { palette, measurements } }) => {
    return css`
      display: flex;
      flex-direction: column;
      background-color: ${palette.common.white};
      box-shadow: 4px 4px 12px -4px ${rgba(palette.common.black, 0.5)};
      border-radius: 8px;

      &:not(:last-of-type) {
        margin-bottom: ${measurements.medium};
      }
    `
  }
)

export const ActionsRow = styled.div<WithTheme & { open: boolean }>(
  ({ theme: { measurements, palette }, open }) => {
    return css`
      border-bottom: ${open ? `1px solid ${palette.grey[500]};` : '0;'};
      margin-bottom: ${measurements.medium};
      padding-left: ${measurements.medium};
      padding-right: ${measurements.medium};
      padding-bottom: ${open ? measurements.medium : 0};
      display: flex;
      flex-direction: row;
      justify-content: flex-end;
      align-items: center;
    `
  }
)
