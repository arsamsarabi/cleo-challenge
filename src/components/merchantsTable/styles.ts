import styled, { css } from 'styled-components'

export const Wrapper = styled.main<WithTheme>(({ theme: { palette } }) => {
  return css`
    flex: 1;
    display: flex;
    flex-direction: column;
    background-color: ${palette.grey[50]};
  `
})

export const PageHeader = styled.div<WithTheme>(
  ({ theme: { palette, measurements } }) => {
    return css`
      display: flex;
      justify-content: flex-start;
      align-items: center;
      margin-bottom: ${measurements.medium};
      h1 {
        font-size: 1.5rem;
        color: ${palette.blue};
      }
      p {
        font-size: 0.9rem;
        color: ${palette.grey[700]};
        margin-left: ${measurements.small};
        span {
          font-weight: bold;
        }
      }
    `
  }
)
