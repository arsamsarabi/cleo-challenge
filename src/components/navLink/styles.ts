import styled, { css } from 'styled-components'
import { Link as RouterLink } from 'react-router-dom'

type LinkProps = WithTheme & {
  isactive: number
}

export const Link = styled(RouterLink)(
  ({ theme: { measurements, palette }, isactive }: LinkProps) => {
    const activeStyles = css`
      color: ${palette.grey[900]};
    `

    return css`
      width: 100%;
      height: 48px;
      display: flex;
      justify-content: flex-start;
      align-items: center;
      padding: ${measurements.small};
      text-decoration: none;
      color: ${palette.grey[600]};
      transition: 0.3s color linear;

      &:not(:last-of-type) {
        margin-bottom: ${measurements.small};
      }

      &:hover {
        ${activeStyles}
      }

      ${isactive &&
      css`
        ${activeStyles}
      `}
    `
  },
)

export const Label = styled.p(({ theme: { measurements, palette } }: WithTheme) => {
  return css`
    margin-left: ${measurements.small};
    font-size: 24px;
  `
})
