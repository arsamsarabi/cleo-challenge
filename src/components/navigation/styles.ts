import styled, { css } from 'styled-components'

export const Wrapper = styled.nav(({ theme: { measurements, palette } }: WithTheme) => {
  return css`
    width: 320px;
    padding: ${measurements.large};
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
  `
})

export const Logo = styled.img(() => {
  return css`
    width: 160px;
  `
})
