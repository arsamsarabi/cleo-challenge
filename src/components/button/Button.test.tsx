import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'

import { render } from '../../utils'
import { ActionButton } from './Button'
import { theme } from '../../styles'

describe('<ActionButton />', () => {
  test('Should have correct default css rules', () => {
    const TEST_ID = 'action-button'
    const { getByTestId, container } = render(
      <ActionButton data-testid={TEST_ID} />
    )
    const styledComponent = getByTestId(TEST_ID)

    expect(styledComponent).toHaveStyleRule('color', theme.palette.blue)

    expect(container.firstChild).toMatchSnapshot()
  })

  test('Should have correct success css rules', () => {
    const TEST_ID = 'action-button'
    const { getByTestId, container } = render(
      <ActionButton data-testid={TEST_ID} mode="success" />
    )
    const styledComponent = getByTestId(TEST_ID)

    expect(styledComponent).toHaveStyleRule('color', theme.palette.common.green)

    expect(container.firstChild).toMatchSnapshot()
  })

  test('Should have correct danger css rules', () => {
    const TEST_ID = 'action-button'
    const { getByTestId, container } = render(
      <ActionButton data-testid={TEST_ID} mode="danger" />
    )
    const styledComponent = getByTestId(TEST_ID)

    expect(styledComponent).toHaveStyleRule('color', theme.palette.common.red)

    expect(container.firstChild).toMatchSnapshot()
  })
})
