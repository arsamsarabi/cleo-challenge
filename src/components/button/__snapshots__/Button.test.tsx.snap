// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionButton /> Should have correct danger css rules 1`] = `
.c0 {
  outline: none;
  color: #c0392b;
  background-color: #ffffff;
  padding: 8px 16px;
  min-height: 40px;
  min-width: 140px;
  text-transform: uppercase;
  border: 2px solid #c0392b;
  border-radius: 4px;
  -webkit-transition: 0.2s all ease-in-out;
  transition: 0.2s all ease-in-out;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 svg {
  color: #c0392b;
  margin-right: 16px;
}

.c0:hover {
  cursor: pointer;
  color: #ffffff;
  background-color: #c0392b;
  font-weight: bold;
  border-color: #c0392b;
}

.c0:hover svg {
  color: #ffffff;
}

<button
  class="c0"
  data-testid="action-button"
  mode="danger"
/>
`;

exports[`<ActionButton /> Should have correct default css rules 1`] = `
.c0 {
  outline: none;
  color: #0815FF;
  background-color: #ffffff;
  padding: 8px 16px;
  min-height: 40px;
  min-width: 140px;
  text-transform: uppercase;
  border: 2px solid #0815FF;
  border-radius: 4px;
  -webkit-transition: 0.2s all ease-in-out;
  transition: 0.2s all ease-in-out;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 svg {
  color: #0815FF;
  margin-right: 16px;
}

.c0:hover {
  cursor: pointer;
  color: #ffffff;
  background-color: #0815FF;
  font-weight: bold;
  border-color: #0815FF;
}

.c0:hover svg {
  color: #ffffff;
}

<button
  class="c0"
  data-testid="action-button"
/>
`;

exports[`<ActionButton /> Should have correct success css rules 1`] = `
.c0 {
  outline: none;
  color: #16a085;
  background-color: #ffffff;
  padding: 8px 16px;
  min-height: 40px;
  min-width: 140px;
  text-transform: uppercase;
  border: 2px solid #16a085;
  border-radius: 4px;
  -webkit-transition: 0.2s all ease-in-out;
  transition: 0.2s all ease-in-out;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 svg {
  color: #16a085;
  margin-right: 16px;
}

.c0:hover {
  cursor: pointer;
  color: #ffffff;
  background-color: #16a085;
  font-weight: bold;
  border-color: #16a085;
}

.c0:hover svg {
  color: #ffffff;
}

<button
  class="c0"
  data-testid="action-button"
  mode="success"
/>
`;
